communication/vehicleconnections/mavsdkstation.h

```diff
@@ -30,12 +33,19 @@ class MavsdkStation : public QObject
 
     QList<QSharedPointer<MavsdkVehicleConnection>> getVehicleConnectionList() const;
 
+private slots:
+    void on_gotHeartbeat(quint8 systemId);
+    void on_timeout();
+
 signals:
     void gotNewVehicleConnection(QSharedPointer<MavsdkVehicleConnection> vehicleConnection);
 
 private:
     mavsdk::Mavsdk mMavsdk;
     QMap<int, QSharedPointer<MavsdkVehicleConnection>> mVehicleConnectionMap;
+
+    QTimer mHeartbeatTimer;
+    QVector<QPair<quint8, int>> mVehicleTimers;
```

Comment:
I think the cleanest solution might be to integrate mVehicleTimers into mVehicleConnectionMap. mVehicleConnectionMap could become a QMap<int, QPair<QSharedPointer<MavsdkVehicleConnection>, int>> but I think it might complicate some other cases (like getting the list of all connections). So I would say it is not worth the effort and we can keep it like this.